http://dev.mutt.org/trac/changeset/298194c414f0

Allow tabs in index subject. Closes #3270.

http://dev.mutt.org/trac/changeset/cff8e8ce4327

Render blanks to spaces in index, fixes [298194c414f0]

wcwidth(<tab>) returns -1 so 'foo<tab>bar' was rendered to 'foobar',
and now to 'foo bar'. Closes #3270 again.


Index: configure.ac
===================================================================
--- configure.ac (revision 5939:199b126a7623)
+++ configure.ac (revision 5958:298194c414f0)
@@ -1216,5 +1216,5 @@
 
 AC_CHECK_HEADERS(wctype.h)
-AC_CHECK_FUNCS(iswalnum iswalpha  iswcntrl iswdigit)
+AC_CHECK_FUNCS(iswalnum iswalpha iswblank iswcntrl iswdigit)
 AC_CHECK_FUNCS(iswgraph iswlower iswprint iswpunct iswspace iswupper)
 AC_CHECK_FUNCS(iswxdigit towupper towlower)
Index: curs_lib.c
===================================================================
--- curs_lib.c (revision 5936:642987450d83)
+++ curs_lib.c (revision 5958:cff8e8ce4327)
@@ -726,5 +726,10 @@
     else
     {
+#ifdef HAVE_ISWBLANK
+      if (iswblank (wc))
+	wc = ' ';
+      else
+#endif
       if (!IsWPrint (wc))
 	wc = '?';
       w = wcwidth (wc);
